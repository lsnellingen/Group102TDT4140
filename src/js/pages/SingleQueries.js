// Libs
import React, { Component } from 'react';
import axios from 'axios';
import UpvoteIcon from 'react-icons/lib/md/thumb-up';

class SingleQueries extends Component {

  constructor(props){
    super(props);
    this.state = {
      answer: ''
    }
  }
  handleOptionChange(field,changeEvent){
    var object = {};
    object[field] = changeEvent.target.value;

    this.setState(object);
  }

  static contextTypes = {
    user: React.PropTypes.object
  };


  render() {

    return (
      <div className="panel panel-info">
      <div className="panel-heading clearfix">
          <div className="col-xs-9 removePadding">
            <span>{this.props.queries.question}</span>
          </div>
        </div>
        <div className="panel-body removePadding">
        <div className="container-fluid">
          <div className="row">
            <div className="col-xs-12 removePadding">
              <ul className="list-group removeMargin">
                <div>
                  <li className="list-group-item"><strong>Question:</strong> {this.props.queries.question}</li>
                </div>
              </ul>
            </div>
          </div>
        </div>
        </div>
      </div>
    );
  }
}




export default SingleQueries;
