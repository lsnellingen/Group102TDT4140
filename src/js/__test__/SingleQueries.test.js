import React from 'react';
import ReactDOM from 'react-dom';
import SingleQueries from '../pages/SingleQueries';
import renderer from 'react-test-renderer';
import { RegistrationForm, LoginLink } from 'react-stormpath';
import {shallow} from 'enzyme';

describe('SingleQueries', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

    test('renders without crashing', () => {


      const tree = shallow(
        <SingleQueries queries={{answers:"Testing answers"}}/>);


      expect(tree).toMatchSnapshot();




    });
});
