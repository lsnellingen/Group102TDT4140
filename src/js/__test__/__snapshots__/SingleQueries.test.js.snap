// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleQueries renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="panel panel-info removePadding"
>
    <div
        className="panel-body removePadding"
    >
        <div
            className="container-fluid"
        >
            <div
                className="row"
            >
                <div
                    className="col-xs-12 removePadding"
                >
                    <ul
                        className="list-group removeMargin"
                    >
                        <div>
                            <li
                                className="list-group-item"
                            >
                                <strong />
                            </li>
                            <select
                                className="form-control"
                                id="alternative-selector"
                                name="alternative-selector"
                                onChange={[Function]}
                                type="alternative-selector"
                            >
                                <option
                                    value="chooseAnswer"
                                >
                                    Choose an answer
                                </option>
                            </select>
                        </div>
                    </ul>
                </div>
            </div>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="panel panel-info removePadding"
>
      <div
            className="panel-body removePadding"
      >
            <div
                  className="container-fluid"
            >
                  <div
                        className="row"
                  >
                        <div
                              className="col-xs-12 removePadding"
                        >
                              <ul
                                    className="list-group removeMargin"
                              >
                                    <div>
                                          <li
                                                className="list-group-item"
                                          >
                                                <strong />
                                          </li>
                                          <select
                                                className="form-control"
                                                id="alternative-selector"
                                                name="alternative-selector"
                                                onChange={[Function]}
                                                type="alternative-selector"
                                          >
                                                <option
                                                      value="chooseAnswer"
                                                >
                                                      Choose an answer
                                                </option>
                                          </select>
                                    </div>
                              </ul>
                        </div>
                  </div>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SingleQueries
        queries={
                Object {
                        "answers": "Testing answers",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SingleQueries {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "user": undefined,
        },
        "props": Object {
          "queries": Object {
            "answers": "Testing answers",
          },
        },
        "refs": Object {},
        "state": Object {
          "answers": "Testing answers",
          "myAlternatives": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="panel panel-info removePadding"
>
          <div
                    className="panel-body removePadding"
          >
                    <div
                              className="container-fluid"
                    >
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12 removePadding"
                                        >
                                                  <ul
                                                            className="list-group removeMargin"
                                                  >
                                                            <div>
                                                                      <li
                                                                                className="list-group-item"
                                                                      >
                                                                                <strong />
                                                                      </li>
                                                                      <select
                                                                                className="form-control"
                                                                                id="alternative-selector"
                                                                                name="alternative-selector"
                                                                                onChange={[Function]}
                                                                                type="alternative-selector"
                                                                      >
                                                                                <option
                                                                                          value="chooseAnswer"
                                                                                >
                                                                                          Choose an answer
                                                                                </option>
                                                                      </select>
                                                            </div>
                                                  </ul>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="panel panel-info removePadding"
>
          <div
                    className="panel-body removePadding"
          >
                    <div
                              className="container-fluid"
                    >
                              <div
                                        className="row"
                              >
                                        <div
                                                  className="col-xs-12 removePadding"
                                        >
                                                  <ul
                                                            className="list-group removeMargin"
                                                  >
                                                            <div>
                                                                      <li
                                                                                className="list-group-item"
                                                                      >
                                                                                <strong />
                                                                      </li>
                                                                      <select
                                                                                className="form-control"
                                                                                id="alternative-selector"
                                                                                name="alternative-selector"
                                                                                onChange={[Function]}
                                                                                type="alternative-selector"
                                                                      >
                                                                                <option
                                                                                          value="chooseAnswer"
                                                                                >
                                                                                          Choose an answer
                                                                                </option>
                                                                      </select>
                                                            </div>
                                                  </ul>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SingleQueries
    queries={
        Object {
            "answers": "Testing answers",
          }
    }
/>,
}
`;
